test/startup.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env sh
+
+while ! nc -z minio 80; do
```

Comment:
This is a common approach to avoid your app container trying to connect to a DB or other external service before it has had a chance to start up.  The test app is going to start making requests as soon as it spins up which will cause it to fail, so we're just waiting until minio is accepting connections before starting the test app.

As for connection pooling and reconnecting, the S3 API is a pseudo-rest api so there are no prolonged open connections.  You could have retries and the like if desired, but right now you'll just get an exception if the S3 server is unreachable.

Reply:
Another example of this: https://github.com/VEuPathDB/service-osi-generator/blob/master/docker/docker-compose.yml#L36-L41